### PR TITLE
[ISSUE 3295]Fix JVM exited when running localbookie with jdk17 (#3295)

### LINKED ISSUE #3295: JVM exited when running localbookie
***Describe the bug***
JVM exited when running localbookie
bk version : latest
jdk version: 
openjdk version "17.0.3" 2022-04-19
OpenJDK Runtime Environment Temurin-17.0.3+7 (build 17.0.3+7)
OpenJDK 64-Bit Server VM Temurin-17.0.3+7 (build 17.0.3+7, mixed mode, sharing)
***To Reproduce***

1. download latest bin zip from release page and upzip it
2. execute `bin/bookkeeper localbookie 10` command
3. JVM can't be created, and got this error message:
```
Unrecognized VM option 'AggressiveOpts'
Error: Could not create the Java Virtual Machine.
Error: A fatal exception has occurred. Program will exit.
```

***Expected behavior***
local bk can run normally
***Screenshots***

image
***Additional context***

I did some some research and I find a [solution](https://github.com/pravega/bookkeeper-operator/blob/master/doc/troubleshooting.md#unrecognized-vm-option), so I add the `-XX:+IgnoreUnrecognizedVMOptions` to `bookkeeper` shell, and problem solved:
image

I wonder can this parameter be added to standard options? a update to doc would save others time too.

### PR DESCRIPTION
**Pull request by weiping-code**
Fix #3295
### Motivation
Currently, localbookie can't run with jdk17

### Changes
- remove flag AggressiveOpts
- add flag AutoBoxCacheMax=20000
- add flag BiasedLockingStartupDelay=500

### CODE REVIEW

**Review by Shoothzj — APPROVED** [review]

**Comment by Shoothzj** [comment]
@sudoHackIn @eolivelli PTAL

File: bin/common.sh [hunk 1]
```diff
@@ -96,7 +96,8 @@ else
     -XX:MaxGCPauseMillis=10 \
     -XX:+ParallelRefProcEnabled \
     -XX:+UnlockExperimentalVMOptions \
-    -XX:+AggressiveOpts \
+    -XX:AutoBoxCacheMax=20000 \
```
Comment (eolivelli) [inline bin/common.sh:99]: I believe that we should drop this super specific values, and let the JVM auto tune.
there is nothing specific in BK that needs this stuff

**Review by eolivelli — CHANGES_REQUESTED** [review]
Thanks for your contribution.
I left a comment, can you please take a look ?

**Comment by Shoothzj** [comment]
@weiping-code please rebase your code, we have deleted gradle build scripts.

File: bin/common.sh (hunk 1 above)
Comment (weiping-code) [reply bin/common.sh:99]: This flag (-XX:+AggressiveOpts) was deprecated in JDK 11 and removed in JDK 12 (https://bugs.java.com/bugdatabase/view_bug.do?bug_id=8199778), and according to link

The only effect that the flag currently has is setting AutoBoxCacheMax = 20000 and BiasedLockingStartupDelay = 500.
Both can be done manually by setting the corresponding flags on the command line.

**Comment by Shoothzj** [comment]
@eolivelli I have addressed the comments. PTAL again

**Review by eolivelli — APPROVED** [review]
LGTM

**Comment by eolivelli** [comment]
we should port this to 4.15 branch

**Comment by eolivelli** [comment]
I am not very sure that we need to enforce:
add flag AutoBoxCacheMax=20000
add flag BiasedLockingStartupDelay=500

I would let the JDK use its own ergonomics
WDYT ?

**Comment by Shoothzj** [comment]
@eolivelli We don't use Integer or locks a lot. So I think remove `AggressiveOpts` is fine.

**Comment by Shoothzj** [comment]
@weiping-code Thanks for your contribution, and welcome to bookkeeper community.

**Pull request merged**